interactive_demo.py

```diff
@@ -0,0 +1,200 @@
+
+        self._input_port = self.DeclareVectorInputPort(
+            'input', BasicVector_[float](num_inputs))
+
+        self.DeclareVectorOutputPort(
+            'output', BasicVector_[float](num_outputs), self._pad_output)
+
+    def _pad_output(self, context, output):
+        in_vec = self._input_port.Eval(context)
+        vector = [v for v in in_vec]
+        for i in range(self._difference):
+            vector.append(self._output_value)
+        output.SetFromVector(vector)
+
+
+def main():
+    builder = DiagramBuilder()
+
+    station = builder.AddSystem(ManipulationStation())
+    station.SetupClutterClearingStation()
+    station.Finalize()
+
+    # Create robot description with re-written paths
+    this_dir = os.path.abspath(os.path.dirname(__file__))
+    sdf_file_path = os.path.join(this_dir, 'iiwa14_no_collision.sdf')
+    with open(os.path.join(this_dir, 'iiwa14_no_collision.sdf.in'), 'r') as file_in:
+        with open(sdf_file_path, 'w') as file_out:
+            file_out.write(file_in.read().replace('PWD_GOES_HERE', this_dir))
+
+    joint_names = [
+        'iiwa_joint_1',
+        'iiwa_joint_2',
+        'iiwa_joint_3',
+        'iiwa_joint_4',
+        'iiwa_joint_5',
+        'iiwa_joint_6',
+        'iiwa_joint_7']
+    joints = []
+    for name in joint_names:
+        joints.append(station.get_controller_plant().GetJointByName(name))
+
+    rclpy.init()
+    node = rclpy.create_node('interactive_demo')
+
+    # Publish SDF content on robot_description topic
+    latching_qos = QoSProfile(depth=1, durability=DurabilityPolicy.TRANSIENT_LOCAL)
+    description_publisher = node.create_publisher(StringMsg, 'robot_description', qos_profile=latching_qos)
+    msg = StringMsg()
+    with open(sdf_file_path, 'r') as sdf_file:
+        msg.data = sdf_file.read()
+    description_publisher.publish(msg)
+
+    clock_system = SystemClock()
+
+    builder.AddSystem(clock_system)
+
+    # Transform broadcaster for TF frames
+    tf_broadcaster = TransformBroadcaster(node)
+
+    # Connect to system that publishes TF transforms
+    tf_system = builder.AddSystem(TFPublisher(tf_broadcaster=tf_broadcaster, joints=joints))
+    tf_buffer = Buffer(node=node)
+    tf_listener = TransformListener(tf_buffer, node)
+    server = InteractiveMarkerServer(node, 'joint_targets')
+    joint_target_system = builder.AddSystem(MovableJoints(server, tf_buffer, joints))
+
+    fk_system = builder.AddSystem(ForwardKinematics(station.get_controller_plant()))
+
+    # Blindly assume first joints in plant are the iiwa14
+    just_arm_joints = builder.AddSystem(
+        PadVector(len(joints), station.get_controller_plant().num_joints(), 0.0))
+
+    builder.Connect(
+        station.GetOutputPort('iiwa_position_measured'),
+        fk_system.GetInputPort('joint_positions')
+    )
+
+    builder.Connect(
+        fk_system.GetOutputPort('transforms'),
+        tf_system.GetInputPort('body_poses')
+    )
+    builder.Connect(
```

Comment:
I don't see how `clock` is used in the `TFPublisher` system

Reply:
`TFPublisher` gets the time from the clock input port and [sets the time of the transform from that](https://github.com/sloretz/drake_ros2_demos/blob/f70382b5c790b1aa2b7dc855fcfd70d4dbdeac0c/drake_ros/drake_ros/systems/tf_publisher.py#L67). This allows it to support both [system time](https://github.com/sloretz/drake_ros2_demos/blob/master/drake_ros/drake_ros/systems/system_clock.py) and [simulated time](https://github.com/sloretz/drake_ros2_demos/blob/master/drake_ros/drake_ros/systems/simulator_clock.py).

Reply:
Yup, missed it somehow.